filebeat/harvester/log.go

```diff
@@ -47,9 +50,42 @@ func NewHarvester(
 	return h, nil
 }
 
+func createLineReader(
+	in FileSource,
+	codec encoding.Encoding,
+	bufferSize int,
+	maxBytes int,
+	readerConfig logFileReaderConfig,
+	mlrConfig *config.MultilineConfig,
+) (processor.LineProcessor, error) {
+	var p processor.LineProcessor
+	var err error
+
+	fileReader, err := newLogFileReader(in, readerConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	p, err = processor.NewLineSource(fileReader, codec, bufferSize)
```

Comment:
it converts the encoding.LineReader (the source) into a LineProcessor adding the timestamps. Kind of an adapter.